engine/src/main/java/org/terasology/world/chunks/remoteChunkProvider/RemoteChunkProvider.java

```diff
@@ -81,12 +81,12 @@ public RemoteChunkProvider(BlockManager blockManager, LocalPlayer localPlayer) {
                 ))
                 .addStage(ChunkTaskProvider.create("", chunk -> {
                     listener.onChunkReady(chunk.getPosition());
-                    worldEntity.send(new OnChunkLoaded(chunk.getPosition()));
```

Comment:
You've moved the call to the worldEntity until after markReady, but what of the `listener.onChunkReady` before it? Should `onChunkReady` also be called only after the chunk is marked ready?

Reply:
You're right. That also looks like it should be moved.

Reply:
The only actual implementation of `onChunkReady` is in `ServerImpl`, and that does assume that the chunk can already be written to.